src/lib.rs

```diff
@@ -101,6 +115,27 @@ where
         // can't know a lower bound, due to the predicate
         (0, upp)
     }
+
+    #[inline]
+    fn fold<B, F>(mut self, mut accum: B, mut f: F) -> B
+    where
+        F: FnMut(B, I::Item) -> B,
+    {
+        if !self.flag {
+            while let Some(x) = self.iter.next_if(&mut self.predicate) {
+                accum = f(accum, x);
+            }
+            //self.flag = true;
+        }
+        accum
+    }
+}
+
+impl<I, P> FusedIterator for PeekingTakeWhile<'_, I, P>
+where
+    I: FusedIterator,
+    P: FnMut(&I::Item) -> bool,
+{
```

Comment:
And then let's change this to be fused only if the nested iterator is fused.

Reply:
huh, that's already the case.

Reply:
Oops, wasn't reading closely enough here, nvm about this comment.

Reply:
I think the implementation without `flag` would in fact not satisfy the `FusedIterator` contract, because the `predicate` may do weird stuff... And the implementation with `flat` would always satisfy the `FusedIterator` contract, because it explicitly fuses it...

Reply:
I added the `not_fused` test case, which demonstrates it.